processor/resourcedetectionprocessor/internal/gcp/gce/gce.go

```diff
@@ -0,0 +1,116 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package gce provides a detector that loads resource information from
+// the GCE metatdata
+package gce // import "cloud.google.com/go/compute/metadata"
+
+import (
+	"context"
+
+	"go.opentelemetry.io/collector/component/componenterror"
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.opentelemetry.io/collector/translator/conventions"
+)
+
+const TypeStr = "gce"
+
+const (
+	typeKey          = "type"
+	hostType         = "host"
+	cloudProviderGCP = "gcp"
+)
+
+type Detector struct {
+	metadata gceMetadata
+}
+
+func NewDetector() *Detector {
+	return &Detector{metadata: &gceMetadataImpl{}}
+}
+
+func (d *Detector) Detect(context.Context) (pdata.Resource, error) {
+	res := pdata.NewResource()
+	res.InitEmpty()
+
+	if !d.metadata.OnGCE() {
+		return res, nil
+	}
+
+	attr := res.Attributes()
+
+	// TODO confirm how to best handle this as we have both "cloud" & "host" resource info
+	attr.InsertString(typeKey, hostType)
```

Comment:
The issue here is that we have resource info for "host" & "cloud", so not sure which type to use.

Is resource type still relevant in OpenTelemetry or can we just ignore this?

Reply:
I don't think there's a `type` label declared in semantic conventions is there?

Reply:
I was confused because `Type` was in the OCensus resource and the [Otel semantic conventions doc](https://github.com/open-telemetry/opentelemetry-specification/tree/master/specification/resource/semantic_conventions#service) refers to resource type, e.g. `type: service`, `type: telemetry.sdk`, although not as an attribute. I'll remove this.